src/java/org/apache/cassandra/io/sstable/SSTableHeaderFix.java

```diff
@@ -84,7 +84,7 @@ public static void fixNonFrozenUDTIfUpgradeFrom30()
         CassandraVersion previousVersion = new CassandraVersion(previousVersionString);
         if (previousVersion.major != 3 || previousVersion.minor > 0)
```

Comment:
todo, this also applies to 3.1 to 3.5
rather than support that, we should probably just add an assertion here that we're not upgrading from those versions

Reply:
The safest thing we could do is to reject `ma` sstables as we would never know whether they are fixed or not. For the safety, the user needs to convert such tables manually with its knowledge of the history

Reply:
Update: 

based on agreement to drop support for `big` sstables before `me` (which makes sense as HCD-1.1 only needs to support upgrades from 3.0.25+, 3.11.11+ and DSE 5.1)

- `BigFormat.earliest_supported_version` raised to `me`
     - need to investigate if scrub honours this too, and that means `SSTableHeaderFix` can be deleted altogether,
- `SSTableHeaderFix` is no longer called from `CassandraDaemon`.
- add startup checks that we're not upgrading from <3.0.25 or <3.11.11